wasm-schema-sandbox/src/lib.rs

```diff
@@ -193,9 +194,9 @@ pub fn validate(
 
     log!("validation complete!");
 
-    let violations = match &result {
+    let violations: Vec<&Violation> = match &result {
         Ok(_) => vec![],
-        Err(violation) => violation.flattened_violations(),
+        Err(violation) => violation.flattened_violations().into_iter().collect(),
```

Comment:
Doesn't `flattened_violations()` already return a `Vec<&Violation>`?
```suggestion
        Err(violation) => violation.flattened_violations(),
```